CLI: Support of MacOS M1/Apple Silicon Build

During my task #2936 I tried out whether it is now possible to also build a CLI for ARM based Macs (M1, Apple Silicon). Unfortunately, it failed with the following logs:

```
/Users/runner/hostedtoolcache/go/1.16.3/x64/pkg/tool/darwin_amd64/compile -o $WORK/b771/_pkg_.a -trimpath "$WORK/b771=>" -shared -p github.com/spf13/viper -lang=go1.12 -complete -buildid NIrZbekVAtuxdfBeojs5/NIrZbekVAtuxdfBeojs5 -goversion go1.16.3 -D "" -importcfg $WORK/b771/importcfg -pack /Users/runner/go/pkg/mod/github.com/spf13/viper@v1.7.0/flags.go /Users/runner/go/pkg/mod/github.com/spf13/viper@v1.7.0/util.go /Users/runner/go/pkg/mod/github.com/spf13/viper@v1.7.0/viper.go
/Users/runner/hostedtoolcache/go/1.16.3/x64/pkg/tool/darwin_amd64/buildid -w $WORK/b487/_pkg_.a # internal
cp $WORK/b487/_pkg_.a /Users/runner/Library/Caches/go-build/8d/8d862a4f36a9433e8ed99a13e152ed70127275583617dfcfb7cb398bc5ab721a-d # internal
github.com/keptn/keptn/cli/pkg/helm
mkdir -p $WORK/b486/
cat >$WORK/b486/importcfg << 'EOF' # internal
# import config
packagefile errors=$WORK/b004/_pkg_.a
packagefile fmt=$WORK/b020/_pkg_.a
packagefile github.com/keptn/keptn/cli/pkg/logging=$WORK/b483/_pkg_.a
packagefile github.com/keptn/kubernetes-utils/pkg=$WORK/b197/_pkg_.a
packagefile helm.sh/helm/v3/pkg/action=$WORK/b487/_pkg_.a
packagefile helm.sh/helm/v3/pkg/chart=$WORK/b207/_pkg_.a
packagefile helm.sh/helm/v3/pkg/kube=$WORK/b683/_pkg_.a
packagefile helm.sh/helm/v3/pkg/release=$WORK/b745/_pkg_.a
packagefile helm.sh/helm/v3/pkg/storage=$WORK/b748/_pkg_.a
packagefile helm.sh/helm/v3/pkg/storage/driver=$WORK/b749/_pkg_.a
packagefile io=$WORK/b014/_pkg_.a
packagefile io/ioutil=$WORK/b108/_pkg_.a
packagefile k8s.io/api/apps/v1=$WORK/b272/_pkg_.a
packagefile k8s.io/apimachinery/pkg/util/yaml=$WORK/b316/_pkg_.a
packagefile k8s.io/cli-runtime/pkg/genericclioptions=$WORK/b596/_pkg_.a
packagefile k8s.io/client-go/rest=$WORK/b384/_pkg_.a
packagefile k8s.io/client-go/tools/clientcmd=$WORK/b472/_pkg_.a
packagefile k8s.io/kubectl/pkg/cmd/util=$WORK/b702/_pkg_.a
packagefile net/http=$WORK/b126/_pkg_.a
packagefile os=$WORK/b027/_pkg_.a
packagefile path/filepath=$WORK/b109/_pkg_.a
packagefile strings=$WORK/b039/_pkg_.a
packagefile time=$WORK/b031/_pkg_.a
EOF
/Users/runner/hostedtoolcache/go/1.16.3/x64/pkg/tool/darwin_amd64/compile -o $WORK/b486/_pkg_.a -trimpath "$WORK/b486=>" -shared -p github.com/keptn/keptn/cli/pkg/helm -lang=go1.16 -complete -buildid dChwypPBi1CF7FWWPvej/dChwypPBi1CF7FWWPvej -goversion go1.16.3 -D "" -importcfg $WORK/b486/importcfg -pack ./pkg/helm/helm_helper.go
/Users/runner/hostedtoolcache/go/1.16.3/x64/pkg/tool/darwin_amd64/buildid -w $WORK/b771/_pkg_.a # internal
cp $WORK/b771/_pkg_.a /Users/runner/Library/Caches/go-build/26/262111fb9b1f9eb9cd685522b4198c75c402da7a0b281c21820548c81d2e1616-d # internal
/Users/runner/hostedtoolcache/go/1.16.3/x64/pkg/tool/darwin_amd64/buildid -w $WORK/b486/_pkg_.a # internal
cp $WORK/b486/_pkg_.a /Users/runner/Library/Caches/go-build/0e/0ec3ef316849d162fb61b174795c0bd1d17b36f156b7a61677db1634ccefb789-d # internal
Error compiling Keptn CLI, exiting ...
make: *** [build-cli] Error 1
Error: Process completed with exit code 2.
```

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -507,10 +507,9 @@ jobs:
           - platform: "macOS-11.0" # cross-compiling the CLI for macOS does not work - see https://github.com/keptn/keptn/issues/2738
             GOOS: "darwin"
             GOARCH: "amd64"
-# builds for apple silicon still dont work https://github.com/keptn/keptn/issues/3987
-#          - platform: "macOS-11.0" # cross-compiling the CLI for macOS does not work - see https://github.com/keptn/keptn/issues/2738
-#            GOOS: "darwin"
-#            GOARCH: "arm64"
+          - platform: "macOS-11.0" # cross-compiling the CLI for macOS does not work - see https://github.com/keptn/keptn/issues/2738
+            GOOS: "darwin"
+            GOARCH: "arm64"
           - platform: "ubuntu-20.04"
             GOOS: "windows"
             GOARCH: "amd64"
@@ -560,6 +559,11 @@ jobs:
           if [[ "$GOOS" == "linux" ]]; then
             # enable static build for linux
             export CGO_ENABLED=0
+          fi
+
+          if [[ "$GOOS" == "darwin" ]]; then
+            # enable CGO for compiling docker credential helper natively
+            export CGO_ENABLED=1
           fi
 
           # build the binary using makefile

